### ISSUE TITLE
CCv0: Update kata-containers to use the updated CCv0 fork

### ISSUE THREAD

**Issue by stevenhorsman** [kata-containers/kata-containers#4489]
In order to stay in sync with the `main` branch who are updating the version of containerd to 1.6.6 in #4422, we want to do the same with our CC fork.

Wang Lei has kindly created a PR with the rebase of our fork on 1.6.6 in https://github.com/confidential-containers/containerd/pull/4 but we want to first test it and then update the code to use it in CCv0

The steps this issue will cover are:
- [x] Create a PR to pull in the [source branch](https://github.com/wllenyj/containerd/tree/rebase-1.6.6) of the CC containerd fork PR into kata-containers' `CCv0` branch (including re-vendoring), fix up any incompatibilities and run tests against it.
  - [x] Update `version.yaml` to point to the CC containerd PR's source branch, and update the version
  - [x] Re-vendor the kata runtime to use the new branch. In `src/runtime` run:
    - `go get github.com/containerd/containerd@v1.6.6` to update the base containerd version
    - `go mod edit -replace github.com/containerd/containerd=github.com/wllenyj/containerd@rebase-1.6.6` to pull in Wang Lei's branch
    - Run `make handle_vendor` to update the vendor modules
  - [x] Create a PR from this code and run the tests to check they work
- [x] Once the kata-containers PR tests works we'll merge the `co-co/containerd` PR
- [ ] Update the our kata-containers PR branch to switch back to the `CC-main` branch of our `co-co/containerd` fork 
  - [x] Update version.yaml to point back to the `CC-main` branch of the co-co fork
  - [x] Re-vendor the kata runtime to use the `CC-main` branch. In src/runtime run:
    - `go mod edit -replace github.com/containerd/containerd=github.com/confidential-containers/containerd@CC-main` to pull back in the main CC fork branch
   - Run `make handle_vendor` to update the vendor modules
   - [x] Push those changes to the PR and re-run the tests.
 - Assuming they work we can merge the kata-containers PR

**Comment by stevenhorsman** [comment]
For reference I've manually tested `crictl` changes with `ccv0.sh`:
```
~/ccv0.sh crictl_create_cc_pod
[sudo] password for stevenh:
# Copyright (c) 2021, 2022 IBM Corporation
#
# SPDX-License-Identifier: Apache-2.0
#
metadata:
  #  name:
  name: kata-cc-busybox-sandbox
  uid: bd312a66-0c80-48b8-a796-447f81c32827
  namespace: default
dns_config:
  servers:
    - 8.8.8.8
POD ID              CREATED             STATE               NAME                      NAMESPACE           ATTEMPT             RUNTIME
45373fa188ae7       5 seconds ago       Ready               kata-cc-busybox-sandbox   default             0                   kata
stevenh@ccv0-containerd-crictl1:~$ ~/ccv0.sh crictl_create_cc_container
f05131d6ca3158bf209bcd36202443b19a3b95bcfb6a8a3e3716def12ce70a5f
CONTAINER           IMAGE                                                    CREATED                  STATE               NAME                     ATTEMPT             POD ID              POD
f05131d6ca315       quay.io/kata-containers/confidential-containers:signed   Less than a second ago   Running             kata-cc-busybox-signed   0                   45373fa188ae7       unknown
stevenh@ccv0-containerd-crictl1:~$ ~/ccv0.sh crictl_delete_cc
f05131d6ca3158bf209bcd36202443b19a3b95bcfb6a8a3e3716def12ce70a5f
f05131d6ca3158bf209bcd36202443b19a3b95bcfb6a8a3e3716def12ce70a5f
Stopped sandbox 45373fa188ae705a874ed9c063f22e01d6b24a1c8c89fcd08fe6aededf3c364b
Removed sandbox 45373fa188ae705a874ed9c063f22e01d6b24a1c8c89fcd08fe6aededf3c364b
```

**Comment by stevenhorsman** [comment]
Closed with #4490 merging
